Modules/Core/Common/src/itkImageSourceCommon.cxx

```diff
@@ -32,16 +32,14 @@ ImageRegionSplitterBase::Pointer globalDefaultSplitter;
 const ImageRegionSplitterBase *
 ImageSourceCommon::GetGlobalDefaultSplitter()
 {
+  // thread safe lazy initialization, prevent race condition on
+  // setting, with an atomic set if null.
+  std::lock_guard<std::mutex> lock(globalDefaultSplitterLock);
   if (globalDefaultSplitter.IsNull())
   {
-    // thread safe lazy initialization, prevent race condition on
```

Comment:
I think this should be returned to the way it was previously coded.  locking is required so that only one thread can create the new object, but then after that, no locking is necessary.

Reply:
What's in master is definitely wrong. One thread could hold the lock and be setting `globalDefaultSplitter` while another thread simultaneously is doing the initial unprotected `if (globalDefaultSplitter.IsNull())` I removed.  I can probably instead use `std::call_once` here too...

Reply:
`call_once` added.